packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -480,6 +480,11 @@ export const DatePickerMixin = (subclass) =>
 
       this.$.overlay.addEventListener('opened-changed', (e) => (this.opened = e.detail.value));
 
+      this.$.overlay.addEventListener('vaadin-overlay-escape-press', () => {
```

Comment:
I think we need some generic solution for the case (hitting Esc inside an overlay without the intention to close it).

A couple of examples:
- Using Esc for returning from interaction mode to navigation mode in Grid
- Using Esc to clear the value from a text field

https://user-images.githubusercontent.com/1222264/169252904-7f7d902d-3b76-4371-bb3e-2f31f61dd7ea.mp4